src/internal.c

```diff
@@ -17003,6 +17003,7 @@ int DecodePrivateKey(WOLFSSL *ssl, word16* length)
         }
 
         if (ssl->buffers.keyType == rsa_sa_algo) {
+    #ifndef NO_RSA
```

Comment:
ret is zero from AllocKey above, now we're ignoring RSA or ECC keys if they weren't build in but we are succeeding on return code? That doesn't make sense to me.

Reply:
I assumed that keyType couldn't be set to rsa_sa_algo unless !NO_RSA.
Changed to return NOT_COMPILED_IN in these cases.